OpenSim/Sandbox/MatlabScripts/build_and_simulate_simple_arm.m

```diff
@@ -0,0 +1,109 @@
+
+% Connect the bodies with pin joints. Assume each body is 1 m long.
+shoulder = PinJoint('shoulder', ...
+                    model.getGround(), ...  % Parent body
+                    Vec3(0), ...            % Location in parent
+                    Vec3(0), ...            % Orientation in parent
+                    humerus, ...            % Child body
+                    Vec3(0, 1, 0), ...      % Location in child
+                    Vec3(0));               % Orientation in child
+elbow = PinJoint('elbow', ...
+                 humerus, Vec3(0), Vec3(0), radius, Vec3(0, 1, 0), Vec3(0));
+
+% Add a muscle that flexes the elbow.
+biceps = Millard2012EquilibriumMuscle('biceps', 200, 0.6, 0.55, 0);
+biceps.addNewPathPoint('origin',    humerus, Vec3(0, 0.8, 0));
+biceps.addNewPathPoint('insertion', radius,  Vec3(0, 0.7, 0));
+
+% Add a controller that specifies the excitation of the muscle.
+brain = PrescribedController();
+brain.addActuator(biceps);
+% Muscle excitation is 0.3 for the first 0.5 seconds, then increases to 1.
+brain.prescribeControlForActuator('biceps', StepFunction(0.5, 3, 0.3, 1));
+
+% Add components to the model.
+model.addBody(humerus);    model.addBody(radius);
+model.addJoint(shoulder);  model.addJoint(elbow);
+model.addForce(biceps);
+model.addController(brain);
+
+% Add a console reporter to print the muscle fiber force and elbow angle. The
+% output will be written to the log file (out.log) in the current directory.
+reporter = ConsoleReporter();
+reporter.set_report_time_interval(1.0);
+reporter.addToReport(biceps.getOutput('fiber_force'));
+reporter.addToReport(elbow.getCoordinate().getOutput('value'), 'elbow_angle');
+model.addComponent(reporter);
+
+% Add display geometry.
+bodyGeometry = Ellipsoid(0.1, 0.5, 0.1);
+bodyGeometry.setColor(Vec3(0.5));  % Gray
+
+% Attach an ellipsoid to a frame located at the center of each body.
+humerusCenter = PhysicalOffsetFrame();
+humerusCenter.setName('humerusCenter');
+humerusCenter.setParentFrame(humerus);
+humerusCenter.setOffsetTransform(Transform(Vec3(0, 0.5, 0)));
+humerus.addComponent(humerusCenter);
+humerusCenter.attachGeometry(bodyGeometry.clone());
+
+radiusCenter = PhysicalOffsetFrame();
+radiusCenter.setName('radiusCenter');
+radiusCenter.setParentFrame(radius);
+radiusCenter.setOffsetTransform(Transform(Vec3(0, 0.5, 0)));
+radius.addComponent(radiusCenter);
+radiusCenter.attachGeometry(bodyGeometry.clone());
+
+% Configure the model.
+state = model.initSystem();
+% Fix the shoulder at its default angle and begin with the elbow flexed.
+shoulder.getCoordinate().setLocked(state, true);
+elbow.getCoordinate().setValue(state, 0.5 * pi);
+model.equilibrateMuscles(state);
+
+% Simulate.
+manager = Manager(model);
+manager.setInitialTime(0);
+manager.setFinalTime(10.0);
+manager.integrate(state);
+
+% Display results recorded by ConsoleReporter.
+if (exist('out.log', 'file') ~= 2), error('log file (out.log) not found'); end
+type('out.log');
+fprintf('Success!\n');
```

Comment:
Random outside observation >> I don't think I've ever seen Matlab code that looks so unlike Matlab code. It eerily looks like Python.